datafusion/src/physical_plan/file_format/mod.rs

```diff
@@ -293,6 +293,34 @@ impl PartitionColumnProjector {
         }
     }
 
+    // Creates a RecordBatch with values from the partition_values. Used when no non-partition values are read
+    fn project_empty(
+        &mut self,
+        batch_size: usize,
+        partition_values: &[ScalarValue],
+    ) -> ArrowResult<RecordBatch> {
+        let expected_cols = self.projected_schema.fields().len();
+        if expected_cols != self.projected_partition_indexes.len() {
+            return Err(ArrowError::SchemaError(format!(
+                "Unexepected number of partition values, expected {} but got {}",
+                expected_cols,
+                partition_values.len()
+            )));
+        }
+        let mut cols = Vec::with_capacity(partition_values.len());
```

Comment:
Small style suggestion: I think we can rewrite this to use `.collect()` on an iterator.

Reply:
Yeah we definitely can. I've updated it to a iterator version.